electrumx/server/block_processor.py

```diff
@@ -3009,18 +3007,18 @@ def advance_txs(
                 # in one and the same tx as making a payment. It's not advisable to do so, but it's a valid possibility
 
                 # Check if there were any payments for subrealms in tx
-                payment_tx_hash = self.create_or_delete_subname_payment_output_if_valid(tx_hash, tx, tx_num, height, atomicals_operations_found_at_inputs, atomicals_spent_at_inputs,  b'spay', self.subrealmpay_data_cache, self.get_expected_subrealm_payment_info, False)
-                if payment_tx_hash:
+                subrealm_payment_tx_hash = self.create_or_delete_subname_payment_output_if_valid(tx_hash, tx, tx_num, height, atomicals_operations_found_at_inputs, atomicals_spent_at_inputs,  b'spay', self.subrealmpay_data_cache, self.get_expected_subrealm_payment_info, False)
+                if subrealm_payment_tx_hash:
                     self.logger.info(f'advance_txs: found valid subrealm payment create_or_delete_subname_payment_output_if_valid {hash_to_hex_str(tx_hash)}')
-                    append_hashX(double_sha256(payment_tx_hash))
+                    append_hashX(double_sha256(subrealm_payment_tx_hash))
                     self.put_op_data(tx_num, tx_hash, "payment-subrealm")
```

Comment:
Is it necessary to write the corresponding operation based on the successful payment status of payment? Like `payment-subrealm` and `payment-subrealm-failed`